services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/pulsar/UnicastFunction.java

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.dataplane.client.pulsar;
+
+import com.futurewei.alcor.web.entity.dataplane.UnicastGoalStateByte;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.functions.api.Context;
+import org.apache.pulsar.functions.api.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UnicastFunction implements Function<UnicastGoalStateByte, UnicastGoalStateByte> {
```

Comment:
Like this function implementation, which is neat.

Reply:
btw, where is this UnicastFunction used?

Reply:
Both UnicastFunction and MulticastFuncation will be bundled as jar packages. These will be deployed to the Pulsar cluster as function workers by the Pulsar CLI tool.

Reply:
I'm trying to use Java interface to deploy function workers. This will enable automatic deployment when the dpm service starts.

Reply:
Cool looking forward to it 👍